django/db/backends/__init__.py

```diff
@@ -823,6 +823,14 @@ def cache_key_culling_sql(self):
         """
         return "SELECT cache_key FROM %s ORDER BY cache_key LIMIT 1 OFFSET %%s"
 
+    def case_cast_sql(self, output_field):
```

Comment:
I think `unification_cast_sql` is a better name since it implies usefulness elsewhere. Adding a link to http://www.postgresql.org/docs/9.4/static/typeconv-union-case.html in the comments of the postgres implementation will also be useful.